packages/matchbox/src/components/TextField/TextField.tsx

```diff
@@ -57,8 +107,18 @@ const FieldBox = React.forwardRef<HTMLInputElement, FieldBoxProps>(function Fiel
         lineHeight={lineHeight}
         py={py}
         required={required}
-        ref={userRef}
+        ref={assignRefs}
       />
+      {type == 'number' && !disabled && (
+        <StyledNumberControls $disabled={disabled}>
```

Comment:
In a native `input type="number"` element, the numbers aren't focusable, so I'm inclined to hide these from keyboard navigation